sycl/unittests/scheduler/InOrderQueueHostTaskDeps.cpp

```diff
@@ -0,0 +1,85 @@
+//==-------- InOrderQueueHostTaskDeps.cpp --- Scheduler unit tests ---------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "SchedulerTest.hpp"
+#include "SchedulerTestUtils.hpp"
+
+#include <helpers/CommonRedefinitions.hpp>
+#include <helpers/PiImage.hpp>
+#include <helpers/PiMock.hpp>
+
+#include <detail/event_impl.hpp>
+#include <detail/queue_impl.hpp>
+
+#include <gtest/gtest.h>
+
+class TestKernel;
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace detail {
+template <> struct KernelInfo<TestKernel> {
+  static constexpr unsigned getNumParams() { return 0; }
+  static const kernel_param_desc_t &getParamDesc(int) {
+    static kernel_param_desc_t Dummy;
+    return Dummy;
+  }
+  static constexpr const char *getName() { return "TestKernel"; }
+  static constexpr bool isESIMD() { return false; }
+  static constexpr bool callsThisItem() { return false; }
+  static constexpr bool callsAnyThisFreeFunction() { return false; }
+};
+
+} // namespace detail
+} // namespace sycl
+} // __SYCL_INLINE_NAMESPACE(cl)
+
+using namespace sycl;
+
+size_t GEventsWaitCounter = 0;
+
+inline pi_result redefinedEventsWait(pi_uint32 num_events,
+                                     const pi_event *event_list) {
+  if (num_events > 0) {
+    GEventsWaitCounter++;
+  }
+  return PI_SUCCESS;
+}
+
+TEST_F(SchedulerTest, InOrderQueueHostTaskDeps) {
+  default_selector Selector;
+  platform Plt{default_selector()};
+  if (Plt.is_host()) {
+    std::cout << "Not run due to host-only environment\n";
+    return;
+  }
+
+  unittest::PiMock Mock{Plt};
+  setupDefaultMockAPIs(Mock);
+  Mock.redefine<detail::PiApiKind::piEventsWait>(redefinedEventsWait);
+
+  context Ctx{Plt};
+  queue InOrderQueue{Ctx, Selector, property::queue::in_order()};
+
+  kernel_bundle KernelBundle =
+      sycl::get_kernel_bundle<sycl::bundle_state::input>(Ctx);
+  auto ExecBundle = sycl::build(KernelBundle);
```

Comment:
What is the point for using `kernel_bundle` in this test?

Reply:
For some reason mock images are not recognized without kernel bundles, and debugging this issue is not the purpose of this patch.

Reply:
Hm, there is no global var which registers a mock image, so it is strange that there is a `kernel_bundle` which defines a kernel with `TestKernel` name, probably another source has a kernel with this name.
I believe we need to register a mock image in this source as well.
Am I missing something?

Reply:
You're right, it was taken from the other source. I added a dedicated image for this particular test.